test_fakeredis.py

```diff
@@ -1797,6 +1797,64 @@ def test_zadd_multiple(self):
         self.assertEqual(self.redis.zrange('foo', 1, 1),
                          [b'two'])
 
+    @redis3_only
+    def test_zadd_with_nx(self):
+        self.zadd('foo', {'four': 4, 'three': 3})
+        self.assertEqual(self.zadd('foo', {'four': 2, 'three': 1}, nx=True), 0)
+        self.assertEqual(self.redis.zrange('foo', 0, -1), [b'three', b'four'])
+        self.assertEqual(self.zadd('foo', {'four': 2, 'three': 1, 'zero': 0}, nx=True), 1)
+        self.assertEqual(self.redis.zrange('foo', 0, -1), [b'zero', b'three', b'four'])
+        self.assertEqual(self.zadd('foo', {'two': 2, 'one': 1}, nx=True), 2)
+        self.assertEqual(self.redis.zrange('foo', 0, -1), [b'zero', b'one', b'two', b'three', b'four'])
+
+    @redis3_only
+    def test_zadd_with_ch(self):
+        self.zadd('foo', {'four': 4, 'three': 3})
+        self.assertEqual(self.zadd('foo', {'four': 4, 'three': 1}, ch=True), 1)
+        self.assertEqual(self.redis.zrange('foo', 0, -1), [b'three', b'four'])
+        self.assertEqual(self.zadd('foo', {'four': 4, 'three': 3, 'zero': 0}, ch=True), 2)
+        self.assertEqual(self.redis.zrange('foo', 0, -1), [b'zero', b'three', b'four'])
+        self.assertEqual(self.zadd('foo', {'two': 2, 'one': 1}, ch=True), 2)
+        self.assertEqual(self.redis.zrange('foo', 0, -1), [b'zero', b'one', b'two', b'three', b'four'])
+
+    @redis3_only
+    def test_zadd_with_xx(self):
+        self.zadd('foo', {'four': 4, 'three': 3})
+        self.assertEqual(self.zadd('foo', {'four': -4, 'three': -3}, xx=True), 0)
+        self.assertEqual(self.redis.zrange('foo', 0, -1), [b'four', b'three'])
+        self.assertEqual(self.zadd('foo', {'four': 4, 'three': 3, 'zero': 0}, xx=True), 0)
+        self.assertEqual(self.redis.zrange('foo', 0, -1), [b'three', b'four'])
+        self.assertEqual(self.zadd('foo', {'two': 2, 'one': 1}, xx=True), 0)
+        self.assertEqual(self.redis.zrange('foo', 0, -1), [b'three', b'four'])
```

Comment:
Add a check that the scores do get updated.

Reply:
I refactored the tests to use `zrange` with scores, and made them a bit more programmatic, to improve readability. Let me know what you think!

Reply:
I'm happy with that approach. What do you think about using a `namedtuple` rather than a dict with fixed keys to store the update information?

Reply:
Updated to use namedtuples, thanks for the suggestion! Let me know if naming works for you :)